### PR TITLE
Add reset password button

### PR DESCRIPTION
**Pull request by Mustahai**
Fix for Issue #1063 "Forgot password" link on login screen

### CODE REVIEW

File: app/src/main/res/layout-land/activity_login.xml [hunk 1]
```diff
@@ -171,16 +171,37 @@
...
-                    android:layout_centerHorizontal="true"/>
+                    android:layout_marginEnd="@dimen/standard_gap"
+                    android:layout_marginLeft="@dimen/standard_gap"
+                    android:layout_marginRight="@dimen/standard_gap"
+                    android:layout_marginStart="@dimen/standard_gap">
+
+                    <fr.free.nrw.commons.ui.widget.HtmlTextView
+                        android:id="@+id/reset_password_link"
+                        style="?android:textAppearanceSmall"
+                        android:layout_width="wrap_content"
```
Comment (neslihanturan) [inline app/src/main/res/layout-land/activity_login.xml]: This should be "0dp" to make "layout_weight=1" work.

File: app/src/main/res/layout-land/activity_login.xml [hunk 2]
```diff
@@ -171,16 +171,37 @@
...
+                        android:layout_height="wrap_content"
+                        android:layout_marginLeft="@dimen/large_gap"
+                        android:layout_weight="1"
+                        android:layout_marginBottom="@dimen/standard_gap"
+                        android:text="@string/reset_password_link" />
+
+                    <fr.free.nrw.commons.ui.widget.HtmlTextView
+                        android:id="@+id/about_privacy_policy"
+                        style="?android:textAppearanceSmall"
+                        android:layout_width="wrap_content"
```
Comment (neslihanturan) [inline app/src/main/res/layout-land/activity_login.xml]: This should be "0dp" to make "layout_weight=1" work.

File: app/src/main/res/layout-land/activity_login.xml [hunk 3]
```diff
@@ -171,16 +171,37 @@
...
+                        android:layout_height="wrap_content"
+                        android:layout_weight="1"
```
Comment (neslihanturan) [inline app/src/main/res/layout-land/activity_login.xml]: You should add `android:gravity="center_horizontal"` line to make texts centered.

File: app/src/main/res/layout-land/activity_login.xml [hunk 4]
```diff
@@ -171,16 +171,37 @@
...
+                    android:layout_marginRight="@dimen/standard_gap"
+                    android:layout_marginStart="@dimen/standard_gap">
+
+                    <fr.free.nrw.commons.ui.widget.HtmlTextView
+                        android:id="@+id/reset_password_link"
+                        style="?android:textAppearanceSmall"
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:layout_marginLeft="@dimen/large_gap"
+                        android:layout_weight="1"
```
Comment (neslihanturan) [inline app/src/main/res/layout-land/activity_login.xml]: You should add `android:gravity="center_horizontal"` line to make texts centered.

**Comment by neslihanturan** [comment]
Thanks @Mustahai for the PR:). It works but doesn't look very symmetric currently. So, some changes requested. Our expectation from your fixes is:

pr_passwd

**Comment by Mustahai** [comment]
Thank you @neslihanturan for your feedback. I have updated and commited changes according to your feedback. I hope they will be according to your expectations.

**Comment by psh** [comment]
Just curious why the mockup in the original issue wasn't implemented -

image

**Comment by neslihanturan** [comment]
Oh, I wasn't aware of that design @psh , so tried make it better on my previous comments. But IMO the design you shared looks better. Sorry for missing that :/

**Comment by misaochan** [comment]
Hi @Mustahai , thanks for your PR. Indeed we had agreed on using the design that @psh submitted at #1063 , could you please modify to suit that? Cheers. :)

**Comment by Mustahai** [comment]
So I've been working on changing the design to fit @psh 's mockup, and unfortunately I've run into two issues that I hope you will be able to help me solve. This is the current layout on my emulator:

imagem

Now my issues are the following:
1 - I can't seem to be able to remove the underline on the forgot password link. There must be a simple way to do it but I'm not finding it.
2- The privacy policy link is moved up by the keyboard, and ends up inside the login frame, like this:

imagem

I hope to solve both issues with your sugestions.

**Comment by neslihanturan** [comment]
@Mustahai , can you please commit your code that results as you shared? So that we can see whats going on.

**Comment by Mustahai** [comment]
The main issue I'm struggling with is that the login activity is a scroll view, so I can't, as far as I know, just send the policy settings link to the bottom of the screen.

**Comment by maskaravivek** [comment]
Have fixed the merge conflicts. Will merge once the checks pass.

**Comment by neslihanturan** [comment]
Thanks for fixing the conflicts @maskaravivek , but according to discussion above, I have some doubts to merge this as is. What do you suggest @psh , should we wait for updates or merge it as is, since it doesn't look very problematic currently.

**Comment by misaochan** [comment]
How does the UI look currently? Does it still have the overlapping text that is shown in the most recent screenshot?

**Comment by ujjwalagrawal17** [comment]
@maskaravivek I think the issue is already fixed. So we can close this PR.

**Pull request closed without merging**